src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectState.cs

```diff
@@ -345,8 +345,7 @@ public ProjectState WithHostProject(HostProject hostProject)
             throw new ArgumentNullException(nameof(hostProject));
         }
 
-        if (HostProject.Configuration.Equals(hostProject.Configuration) &&
-            HostProject.RootNamespace == hostProject.RootNamespace)
+        if (HostProject.Configuration.Equals(hostProject.Configuration))
```

Comment:
FWIW, since `RazorConfiguration` is a record, it should be fine to use `==` for value equality:

```suggestion
        if (HostProject.Configuration == hostProject.Configuration)
```